### ISSUE TITLE
Fix S3459 FP: for method backed property

### ISSUE THREAD

**Issue by DelphinRP** [SonarSource/sonar-dotnet#4739]
### Description

A property that calls a method for getting its value is flagged by Sonarqube with S3459.

### Repro steps

```C#
using System;

namespace SonarS3459
{
    public class Program
    {
        private int SomeValue { get => GetValue(); }

        static void Main( string[] args )
        {
            var program = new Program();

            Console.WriteLine( $"Hello World: {program.DoStuff()}" );
        }

        public int DoStuff()
        {
            return SomeValue;
        }

        private int GetValue()
        {
            return GetHashCode();
        }
    }
}
```

### Expected behavior

No S3459 reported for `SomeValue`.

### Actual behavior

Sonarqube reports S3459 for `SomeValue`:

![s3459fp](https://user-images.githubusercontent.com/85873523/127503322-2844e5cf-2e1e-4608-9e37-8764342e701f.png)

### Known workarounds

No known workaround.

### Related information

* C#/VB.NET Plugins version: `SonarLint for Visual Studio 2019 (4.37.0.34839)`
* Visual Studio version `16.10.4`
* MSBuild / dotnet version: `5.0.302`
* SonarScanner for .NET version (if used)
* Operating System: `Windows 10`

**Comment by DelphinRP** [comment]
S3459 actually is correct in that case as automatic properties always create a setter that is never used.
Issues is a false false positive.

Closing.

**Comment by ndsvw** [comment]
Hi @DelphinRP,
Could you explain why this is not a valid false positive?
We're having the same issue.
The warning says "auto-property", but ```private int SomeValue { get => GetValue(); }``` is not an auto-property, right? Auto-property sound more like this: ```private int SomeValue { get; set; }```.
Also: What do you mean that there is a setter? You have not specified one. So there shouldn't be any setter. Not even when it's compiled.

**Comment by DelphinRP** [comment]
Hi @ndsvw,

my [explanation](https://github.com/SonarSource/sonar-dotnet/issues/4739#issuecomment-889171554) is indeed wrong. The property is indeed **not** an automatic property.

I still accepted the S3459 as using a `private` property instead of calling the backing method directly seems to be bad practice to me. I ended up replacing my property with calls to the method instead.
S3459 does not appear on properties that are `public` which makes sense to me as it's not the consumer's business to know of the property's inner workings.

That being said, the S3459 makes no sense as a warning in this case. Its description is not correct in this case.
I'm not sure, if it should be removed without replacement or if one wants to show a different rule (e.g. "You might want to replace your private property with direct calls to the method.").

Re-opening the issue.

**Comment by andrei-epure-sonarsource** [comment]
I confirm that the message is wrong and confusing.

Indeed it's a bad practice for a property to just return the result of a method call. I've opened https://github.com/SonarSource/sonar-dotnet/issues/4868 as a rule idea.

**Comment by mary-georgiou-sonarsource** [comment]
This is not reproducible anymore - it seems to be fixed.
